src/ui/compilationStatusView.ts

```diff
@@ -55,26 +60,38 @@ export default class CompilationStatusView {
     this.updateActiveDocumentStatus();
   }
 
+  private areParamsOutdated(params: ICompilationStatusParams): boolean {
+    const previous = this.documentStatusMessages.get(getVsDocumentPath(params));
+    return previous?.version != null && params.version != null
+       && previous.version > params.version;
+  }
+
   private compilationStatusChanged(params: ICompilationStatusParams): void {
+    if(this.areParamsOutdated(params)) {
+      return;
+    }
     this.documentStatusMessages.set(
       getVsDocumentPath(params),
-      toStatusMessage(params.status, params.message)
+      { status: toStatusMessage(params.status, params.message),
+        version: params.version
+      }
```

Comment:
```suggestion
      {
        status: toStatusMessage(params.status, params.message),
        version: params.version
      }
```
Have `status` on a new-line too. 👍